crates/router/src/core/payments/operations/payment_status.rs

```diff
@@ -356,6 +356,23 @@ pub async fn get_payment_intent_payment_attempt(
                     )
                     .await?;
             }
+            api_models::payments::PaymentIdType::PreprocessingId(ref id) => {
```

Comment:
Validate this is required or not?

Reply:
This is required,we have added PreprocessingId variant in PaymentIdType, this function expects  to return payment_intent and payment_attempt.